runtime/executor/src/main/java/edu/snu/nemo/runtime/executor/TaskGroupExecutor.java

```diff
@@ -307,15 +308,26 @@ private void launchOperatorTask(final OperatorTask operatorTask) {
 
     long accumulatedBlockedReadTime = 0;
     long accumulatedWriteTime = 0;
+    long accumulatedSerializedBlockSize = 0;
+    long accumulatedEncodedBlockSize = 0;
+    boolean blockSizeAvailable = true;
     // Consumes all of the partitions from incoming edges.
     for (int srcTaskNum = 0; srcTaskNum < sourceParallelism.get(); srcTaskNum++) {
       try {
         // Because the data queue is a blocking queue, we may need to wait some available data to be pushed.
         final long blockedReadStartTime = System.currentTimeMillis();
-        final Pair<Iterator, String> availableData = dataQueue.take();
+        final Pair<DataUtil.IteratorWithNumBytes, String> availableData = dataQueue.take();
         final long blockedReadEndTime = System.currentTimeMillis();
         accumulatedBlockedReadTime += blockedReadEndTime - blockedReadStartTime;
         transform.onData(availableData.left(), availableData.right());
+        if (blockSizeAvailable) {
+          try {
+            accumulatedSerializedBlockSize += availableData.left().getNumSerializedBytes();
+            accumulatedEncodedBlockSize += availableData.left().getNumEncodedBytes();
+          } catch (final UnsupportedOperationException e) {
```

Comment:
Why don't we add a special exception for this? Throwing general exception and catching it might be dangerous.

Reply:
In my humble opinion, `UnsupportedOperationException` provides good semantics for this situation. Can we take it into offline discussion?